Implemented .zip files uploading.

Anixen opened the pull request:
The addRessource() function in /render/views.py has been broken down in
several function to handle several cases :
-uploading an archive file (.zip only for the moment)
-uploading a single file

In case an extracted file is uploaded, the image files are extracted to
/tmp/, registered into the BD, copied to the /ressources/render/
directory and then removed.

Anixen committed:
Implemented .zip files uploading.

The addRessource() function in /render/views.py has been broken down in
several function to handle several cases :
-uploading an archive file (.zip only for the moment)
-uploading a single file

In case an extracted file is uploaded, the image files are extracted to
/tmp/, registered into the BD, copied to the /ressources/render/
directory and then removed.

Olivier-Daire commented:
Trying to upload a zip file containing two pictures from `http://localhost:5005/upload`, this is what I get : 

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1478, in full_dispatch_request
    response = self.make_response(rv)
  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1566, in make_response
    raise ValueError('View function did not return a response')
ValueError: View function did not return a response
```

Anixen committed:
Fixed a ViewFunction error caused by indentation.

Anixen committed:
Attempting to fix uploading archives via Firefox by recognizing more mimetypes

Anixen committed:
New attempt to fix mimetype guessing on uploading ressources.

We do not rely anymore on the mimetype given via the POST request, but
instead guess frome the filename via mimetype.guess_type()

Anixen committed:
Quick fix : gues_types returns a tuple, not a string.

Anixen committed:
Cleaning up code and (re)moving some commented lines

Anixen committed:
Merge branch 'develop' into upload_archive
